### ISSUE TITLE
New warning in tests

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1905]
Seen in https://github.com/r-lib/lintr/pull/1879
For log, see https://github.com/r-lib/lintr/actions/runs/4041114622/jobs/6949645553

```r
══ Failed tests ════════════════════════════════════════════════════════════════
── Error (test-get_source_expressions.R:111): Warns if encoding is misspecified ──
Error in `grepl(pat, text, perl = TRUE)`: (converted from warning) input string 3 is invalid UTF-8
Backtrace:
    ▆
 1. └─lintr::lint(filename = file, parse_settings = FALSE) at test-get_source_expressions.R:111:2
 2.   └─lintr::get_source_expressions(filename, lines)
 3.     └─lintr:::extract_r_source(...)
 4.       └─lintr:::get_knitr_pattern(filename, lines)
 5.         └─("knitr" %:::% "detect_pattern")(...)

[ FAIL 1 | WARN 0 | SKIP 1 | PASS 4106 ]
```

**Comment by IndrajeetPatil** [comment]
Coming from here, I think:

https://github.com/r-lib/lintr/blob/c69f7926e2b01d06171dfe63bef6a90fd9a656f8/tests/testthat/dummy_projects/project/cp1252.R#L1-L4

**Comment by IndrajeetPatil** [comment]
Closing in favour of https://github.com/r-lib/lintr/issues/1920
